### DISCUSSION TITLE
Best practices for early returns

### DISCUSSION THREAD

**Discussion by ineu** [nushell/nushell#8147] [Q&A] [answered]
So there's a def validating the input and returning early. It's obvious while you don't need the data from the def:
```
def foo [] {
  if $condition { return }
  # go on
}
```

But what if I want to return something useable? The following exits early and prints to stdout, but doesn't actually *return* anything to the caller:
```
def foo [] {
  if $condition { ls; return }
  # go on
}
```
I need to do either
```
def foo [] {
  if $condition { return (ls) }
  # go on
}
```
or
```
def foo [] {
  if $condition { ls | return $in }
  # go on
}
```

My questions are:
1. Is there another approach except the two above?
2. Which of the two is preferrable? Are there some coding standards/best practices regarding this case?

Thank you

**Comment by fdncred** [comment, marked as answer]
I'm not sure we have any best practices yet. We just added the `return` command a few releases ago.
